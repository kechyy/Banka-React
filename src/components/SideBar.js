import React, {Component} from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import AccountItem from './AccountItem';
import { getTransactionAction } from '../store/modules/transaction/actions';

class SideBar extends Component {
  constructor() {
    super();
    this.state= { status:true, topMenu:true }
    this.acctToggle = this.acctToggle.bind(this);
    this.getTransaction = this.getTransaction.bind(this);
  }
  acctToggle() {
    this.setState(prevState => ({
      status: !prevState.status
    }))
  }
  getTransaction(acctNumber) {
    this.props.getTransactionAction(acctNumber);
  }
  render() {

    const changeState = this.state.status?'none':'block';
    const menuToggle = this.state.topMenu?'none':'block';
    return (
      <div className="left mainSideBar">
              <div className="row">
                <div className="col-8-lg sideBar">
                  <aside>
                    <div><img src="../images/logo2.png" alt="" className="logo" /></div>   
                    <div className="sideBarTitle">Client User</div>
                    <ul className="sideMenu">
                      <li className="userProfiles">
                        - &nbsp; 
                        <Link to='/user-profile' data-title="Profile Details" >Profile</Link>
                      </li>
                      <li className="accountForms"><span className="fa fa-book" /> 
                        <Link to='/create-account' data-title="Create Account" >Create Bank Account</Link>
                      </li>
                      <li className="transact">- &nbsp; 
                      <a onClick={this.acctToggle} >Transaction History</a>
                        <ul className="sideBarDropDown" style={{display:changeState}}>
                          { this.props.account.map(accounts => {
                            return <AccountItem accountList={accounts} getTransaction={this.getTransaction} />
                          }) }
                        </ul> 
                      </li>
                    </ul>     
                  </aside>
                </div>
              </div>
            </div>
    )
  }
}
const mapStateToProps = state => ({
  userProfile: state.profileReducer.profile,
  account: state.accountReducer.account
});

const mapDispatchToProps = {
  getTransactionAction
}
export default connect(mapStateToProps, mapDispatchToProps)(SideBar);