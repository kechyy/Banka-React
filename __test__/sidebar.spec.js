import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount, shallow } from './enzyme';
import SideBar from '../src/components/SideBar';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { BrowserRouter, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';


const mockStore = configureMockStore([thunk]);
let store;
const initialSate = {
  status: 'rest'
};

const props = {
  isAuthenticated: true,
  profileAction: jest.fn(),
  userProfile: jest.fn()
};

const renderWithEnzymes = state => {
  store = mockStore({
    authReducer: {
      error: null,
      isAuthenticated: false,
      user: {},
      status: 'rest'
    },
    profileReducer: {
      profile: {},
      status: 'rest'
    },
    accountReducer: {
      account: [],
      error: null,
      status: 'rest'
    },
    SideBar: {
      ...state
    }
  });
  return mount(
    <Provider store={store}>
      <Router>
        <SideBar {...props} />
      </Router>
    </Provider>
  );
};


describe('SideBar', () => {
  afterEach(() => {
    jest.clearAllMocks();
    store = null;
  });

  it('should have 1 table on SideBar', () => {
    const wrapper = renderWithEnzymes();
    expect(wrapper.find('ul').length).toBe(2);
   
  });

  it('should render SideBar Component', () => {
    const wrapper = renderWithEnzymes();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});