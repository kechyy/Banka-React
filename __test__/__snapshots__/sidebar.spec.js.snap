// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render SideBar Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SideBar)
        isAuthenticated={true}
        profileAction={[MockFunction]}
        userProfile={[MockFunction]}
      >
        <SideBar
          account={Array []}
          getTransactionAction={[Function]}
          isAuthenticated={true}
          profileAction={[MockFunction]}
          userProfile={Object {}}
        >
          <div
            className="left mainSideBar"
          >
            <div
              className="row"
            >
              <div
                className="col-8-lg sideBar"
              >
                <aside>
                  <div>
                    <img
                      alt=""
                      className="logo"
                      src="../images/logo2.png"
                    />
                  </div>
                  <div
                    className="sideBarTitle"
                  >
                    Client User
                  </div>
                  <ul
                    className="sideMenu"
                  >
                    <li
                      className="userProfiles"
                    >
                      -  
                      <Link
                        data-title="Profile Details"
                        to="/user-profile"
                      >
                        <a
                          data-title="Profile Details"
                          href="/user-profile"
                          onClick={[Function]}
                        >
                          Profile
                        </a>
                      </Link>
                    </li>
                    <li
                      className="accountForms"
                    >
                      <span
                        className="fa fa-book"
                      />
                      <Link
                        data-title="Create Account"
                        to="/create-account"
                      >
                        <a
                          data-title="Create Account"
                          href="/create-account"
                          onClick={[Function]}
                        >
                          Create Bank Account
                        </a>
                      </Link>
                    </li>
                    <li
                      className="transact"
                    >
                      -  
                      <a
                        onClick={[Function]}
                      >
                        Transaction History
                      </a>
                      <ul
                        className="sideBarDropDown"
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                      />
                    </li>
                  </ul>
                </aside>
              </div>
            </div>
          </div>
        </SideBar>
      </Connect(SideBar)>
    </Router>
  </BrowserRouter>
</Provider>
`;
